test/applicationData.js

```diff
@@ -0,0 +1,224 @@
+const chai = require('chai')
+chai.Should()
+chai.use(require('chai-things'))
+chai.use(require('chai-json-equal'))
+const fetch = require('node-fetch')
+const freeport = require('freeport-promise')
+const {
+  startServerP,
+  getReadOnlyToken,
+  getWriteToken,
+  getAdminToken
+} = require('./servertestutilities')
+const fs = require('fs')
+const promisify = require('util').promisify
+const path = require('path')
+
+const APP_ID = 'testApplication'
+const APP_VERSION = '1.0'
+
+const TEST_SETTINGS = {
+  something: 100,
+  sometingElse: 'hello'
+}
+
+describe('Application Data', () => {
+  var server,
+    url,
+    port,
+    adminToken,
+    writeToken,
+    readToken,
+    readHeaders,
+    writeHeaders,
+    adminHeaders
+
+  before(async function () {
+    this.timeout(5000)
+    port = await freeport()
+    url = `http://0.0.0.0:${port}`
+    
+    server = await startServerP(port, true)
+
+    readToken = await getReadOnlyToken(server)
+    writeToken = await getWriteToken(server)
+    adminToken = await getAdminToken(server)
+
+    readHeaders = {
+      Cookie: `JAUTHENTICATION=${readToken}`
+    }
+    writeHeaders = {
+      Cookie: `JAUTHENTICATION=${writeToken}`
+    }
+    adminHeaders = {
+      Cookie: `JAUTHENTICATION=${adminToken}`
+    }
+
+    let paths = [
+      path.join(
+        process.env.SIGNALK_NODE_CONDFIG_DIR,
+        'applicationData',
+        'global'
+      ),
+      path.join(process.env.SIGNALK_NODE_CONDFIG_DIR, 'applicationData', 'user')
+    ]
+    paths.forEach(p => {
+      if (fs.existsSync(p)) {
+        fs.readdirSync(p).forEach(f => {
+          fs.readdirSync(path.join(p, f)).forEach(s => {
+            fs.unlinkSync(path.join(p, f, s))
+          })
+        })
+      }
+    })
+  })
+
+  after(async function () {
+    await server.stop()
+  })
+
+  it('fetch global returns empty data', async function () {
+    var result = await fetch(
+      `${url}/signalk/v1/applicationData/global/${APP_ID}/:${APP_VERSION}`,
+      {
+        headers: readHeaders
+      }
+    )
+    result.status.should.equal(200)
+    let data = await result.json()
+    data.should.jsonEqual({})
+  })
+
+  it('post global fails readonly user', async function () {
+    var result = await fetch(
+      `${url}/signalk/v1/applicationData/global/${APP_ID}/:${APP_VERSION}`,
+      {
+        method: 'POST',
+        headers: {
+          ...readHeaders,
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(TEST_SETTINGS)
+      }
+    )
+    result.status.should.equal(401)
```

Comment:
Should this be 403 Forbidden, not 401 Unauthorized?

Reply:
Probably, but I don't think this PR is the best place to fix it